circuit-foundation/src/commonMain/kotlin/com/slack/circuit/foundation/NavigableCircuitContent.kt

```diff
@@ -103,33 +104,15 @@ public fun <R : Record> NavigableCircuitContent(
 
   CompositionLocalProvider(LocalRetainedStateRegistry provides outerRegistry) {
     decoration.DecoratedContent(activeContentProviders, backStack.size, modifier) { provider ->
-      // We retain the record's retained state registry for as long as the back stack
```

Comment:
Most of this below has been moved into the Record's movable content. This allows us to remember much more of this (as it moves with the content). Currently most of this is dropped as soon as the back stack changes.